### PR TITLE
Added code for validation of vm name

### PR DESCRIPTION
**Pull request by dheerajd-msys**

### CODE REVIEW

File: lib/chef/knife/azurerm_server_create.rb [hunk 1]
```diff
@@ -265,6 +263,14 @@ def validate_params!
...
           raise ArgumentError, "When --azure-vnet-subnet-name is specified, the --azure-vnet-name must also be specified."
         end
+        
+        if !is_image_windows?
+          if (locate_config_value(:azure_vm_name).match /^(?=.*[a-zA-Z-])([a-zA-z0-9-]{1,64})$/).nil? 
+            raise ArgumentError, "VM name can only contain alphanumeric and hyphen(-) characters and maximun length cannot exceed 64 charachters."
+          end
+        elsif (locate_config_value(:azure_vm_name).match /^(?=.*[a-zA-Z-])([a-zA-z0-9-]{1,15})$/).nil?
+          raise ArgumentError, "VM name can only contain alphanumeric and hyphen(-) characters and maximun length cannot exceed 15 charachters."
+        end
```
Comment (NimishaS) [inline lib/chef/knife/azurerm_server_create.rb]: @adamedx , currently azure ARM has this validation on the server name. Our reason for adding this validation at initial level is that user doesn't have to wait too long to get this error from server side.
However if the server side validation changes in future, there will be a mismatch in our validation and server side validation.
Please add your views on this.

**Comment by adamedx** [comment]
Good idea @Aliasgar16, just needs unit tests.

**Comment by dheerajd-msys** [comment]
@adamedx, i have added the Unit tests for various scenarios, let me know if anything else needs to be done here.

**Comment by adamedx** [comment]
:+1:

**Pull request merged**